internal/util/importutilv2/parquet/field_reader.go

```diff
@@ -670,28 +747,162 @@ func parseSparseFloatRowVector(str string) ([]byte, uint32, error) {
+// to return one-dim list. We use the start/end position of ValueOffsets() to get the correct sparse vector
+// from ListValues().
+// Note that arrow.Uint32.Value(int i) accepts an int32 value, the max length of indices/values is max value of int32
+func parseSparseFloatVectorStructs(structs []map[string]arrow.Array) ([][]byte, uint32, error) {
+	byteArr := make([][]byte, 0)
+	maxDim := uint32(0)
+	for _, st := range structs {
+		indices, ok1 := st[sparseVectorIndice]
+		values, ok2 := st[sparseVectorValues]
+		if !ok1 || !ok2 {
+			return nil, 0, merr.WrapErrImportFailed("Invalid parquet struct for SparseFloatVector: 'indices' or 'values' missed")
+		}
+		if indices.Len() != values.Len() {
+			return nil, 0, merr.WrapErrImportFailed("Invalid parquet struct for SparseFloatVector: 'indices' or 'values' length mismatched")
+		}
+		indicesList, ok1 := indices.(*array.List)
+		valuesList, ok2 := values.(*array.List)
+		if !ok1 || !ok2 {
+			return nil, 0, merr.WrapErrImportFailed("Invalid parquet struct for SparseFloatVector: 'indices' or 'values' is not list")
+		}
+
+		// technically, DataType() of array.List must be arrow.ListType, but we still check here to ensure safety
+		indicesListType, ok1 := indicesList.DataType().(*arrow.ListType)
+		valuesListType, ok2 := valuesList.DataType().(*arrow.ListType)
+		if !ok1 || !ok2 {
+			return nil, 0, merr.WrapErrImportFailed("Invalid parquet struct for SparseFloatVector: incorrect arrow type of 'indices' or 'values'")
+		}
+
+		indexDataType := indicesListType.Elem().ID()
+		valueDataType := valuesListType.Elem().ID()
+
+		// The array.Uint32/array.Int64/array.Float32/array.Float64 are derived from arrow.Array
+		// The ListValues() returns arrow.Array interface, but the arrow.Array doesn't have Value(int) interface
+		// To call array.Uint32.Value(int), we need to explicitly cast the ListValues() to array.Uint32
+		// So, we declare two methods here to avoid type casting in the "for" loop
+		type GetIndex func(position int) uint32
+		type GetValue func(position int) float32
+
+		var getIndexFunc GetIndex
+		switch indexDataType {
+		case arrow.INT32:
+			indicesList := indicesList.ListValues().(*array.Int32)
+			getIndexFunc = func(position int) uint32 {
+				return (uint32)(indicesList.Value(position))
+			}
+		case arrow.UINT32:
+			indicesList := indicesList.ListValues().(*array.Uint32)
+			getIndexFunc = func(position int) uint32 {
+				return indicesList.Value(position)
+			}
+		case arrow.INT64:
+			indicesList := indicesList.ListValues().(*array.Int64)
+			getIndexFunc = func(position int) uint32 {
+				return (uint32)(indicesList.Value(position))
+			}
+		case arrow.UINT64:
+			indicesList := indicesList.ListValues().(*array.Uint64)
+			getIndexFunc = func(position int) uint32 {
+				return (uint32)(indicesList.Value(position))
+			}
+		default:
+			return nil, 0, merr.WrapErrImportFailed("Invalid parquet struct for SparseFloatVector: index type must be uint32/int32/uint64/int64")
```

Comment:
please help to print actual index type as well

Reply:
done